### PR TITLE
Make path-matching regexes more strict to reduce spurious matches

### LINKED ISSUE #36: Spurious matches when the "for" keyword is inside a string
The regexes used to match import statements assume that the import-related keywords (`from`, `import`, `require`) always carry their functional meaning. But these keywords can also appear inside of strings and comments, where they no longer mean the same thing.

I stumbled on an example that, stripped down, looks like this:

```js
const testCase = `
    'a string with keyword from '
    // The from keyword in that string can cause
    // a match up to the next quote, since the regex does not
    // know that the keyword is in a string context
    'another string using same quote type'
  `;
```

The regexes end up claiming that the following is an imported path:

```js
`
    // The from keyword in that string can cause
    // a match up to the next quote, since the regex does not
    // know that the keyword is in a string context
`
```

To completely address this we'd need a more elaborate matching mechanism that strips comments and strings before matching keywords.

However, a band-aid fix to the regexes to have stricter matching would resolve cases like this one. In particular, ensuring that import paths are single-line strings will do the trick.

### PR DESCRIPTION
**Pull request by adam-coster**
Closes #36 (with the easy fix)

### CODE REVIEW

File: src/utils/ImportPathResolver.ts [hunk 1]
```diff
@@ -32,15 +32,15 @@ import { resolve, join, dirname } from 'path';
 export type StringReplacer = (importStatement: string) => string;
 
 const anyQuote = '["\']';
-const notQuote = '[^"\']';
-const importString = `(?:${anyQuote}${notQuote}+${anyQuote})`;
+const pathStringContent = `[^"'\r\n]+`;
```
Comment (adam-coster) [inline src/utils/ImportPathResolver.ts:35]: Previously was using `noteQuote` to match characters between two quotes. But this would match anything, even if it couldn't be part of an import path. Renamed based on actual use and included newline chars in the restricted set. This prevents matches to things that definitely aren't valid import paths due to being split across lines or having internal quotes, without assuming any other restrictions on valid path characters.

File: tests/test.spec.ts [hunk 2]
```diff
@@ -17,12 +17,13 @@ var module = require
 (
   '1'
 )  ;
-import module from ('2');
+import module from '2';
```
Comment (adam-coster) [inline tests/test.spec.ts:20]: The fact that parens were allowed made it more difficult to crate a sufficiently restrictive path-matching regex. But having parens around import-paths strings (unless using functional importing, e.g. `import("path")`) is not valid according to spec. I think you can technically do it with CommonJS since imports are synchronous, but it's not allowed with Modules since imports are async.

Long story short: if we don't allow parens around import paths we don't lose any important functionality and restrictive matching is easier.

File: src/utils/ImportPathResolver.ts [hunk 3]
```diff
@@ -32,15 +32,15 @@ import { resolve, join, dirname } from 'path';
...
+const importString = `(?:${anyQuote}${pathStringContent}${anyQuote})`;
 
 // Separate patterns for each style of import statement,
 // wrapped in non-capturing groups,
 // so that they can be strung together in one big pattern.
 const funcStyle = `(?:\\b(?:import|require)\\s*\\(\\s*${importString}\\s*\\))`;
 const globalStyle = `(?:\\bimport\\s+${importString})`;
-const fromStyle = `(?:\\bfrom\\s+${notQuote}*?${importString})`;
+const fromStyle = `(?:\\bfrom\\s+${importString})`;
```
Comment (adam-coster) [inline src/utils/ImportPathResolver.ts:43]: This was the main culprit, mean to allow parens around import paths.

The fact that parens were allowed made it more difficult to crate a sufficiently restrictive path-matching regex. But having parens around import-paths strings (unless using functional importing, e.g. import("path")) is not valid according to spec. I think you can technically do it with CommonJS since imports are synchronous, but it's not allowed with Modules since imports are async.

Long story short: if we don't allow parens around import paths we don't lose any important functionality and restrictive matching is easier.

File: tests/test.spec.ts [hunk 4]
```diff
@@ -70,6 +71,17 @@ it(`Import regex matches import statements`, () => {
...
 
+it(`Import regex does not match edge cases from keywords in strings`, function () {
+  const testCase = `
+    'a string with keyword from '
+    // The from keyword in that string can cause
+    // a match up to the next quote, since the regex does not
+    // know that the keyword is in a string context
+    'another string using same quote type'
+  `;
+  expect(newImportStatementRegex().exec(testCase)?.[0]).toBeUndefined();
```
Comment (adam-coster) [inline tests/test.spec.ts:82]: Stripped down version of the scenario I ran into in production, where the word "for" inside a string caused a spurious multiline string to be treated as an import path. This test was failing until I updated the regexes.
